README.md

```diff
@@ -1,7 +1,434 @@
+    optimizations if it knows that the object category is
+    group: `(&(objectCategory=group)(member={0}))` this may be relevant
+    if using Active Directory's matching rule in chain extension, e.g.
+    `(&(objectCategory=group)(member:1.2.840.113556.1.4.1941:={0}))`
+
+Note: in this field there are two available substitutions:  
+{`0`} - the fully qualified DN of the user  
+{`1`} - the username portion of the user
+
+#### Parse user attribute for list of groups
+
+![](docs/images/Screen_Shot_2014-07-15_at_10.19.14.png){.confluence-embedded-image}  
+Some LDAP servers can provide a `memberOf` attribute within the User's
+record:
+
+-   Active Directory
+-   OpenLDAP with the [memberof
+    overlay](http://www.openldap.org/doc/admin24/overlays.html#Reverse%20Group%20Membership%20Maintenance){.external-link}
+    active (untested, and as memberof is an operational attribute in
+    OpenLDAP it must be explicitly requested, so likely some hacking of
+    LDAPSecurityRealm.groovy required)
+-   (If you know of others please provide details here)
+
+This attribute can be used to simplify the group search and return the
+group membership immediately without a second LDAP query. Note, however,
+that this may result in only direct group membership being supported.
+
+The `group membership attribute field` controls the attribute name that
+is used to determine the groups to which a user belongs.
+
+### Manager DN and Manager Password
+
+If your LDAP server doesn't support anonymous binding (IOW, if your LDAP
+server doesn't even allow a query without authentication), then Jenkins
+would have to first authenticate itself against the LDAP server, and
+Jenkins does that by sending "manager" DN and password.
+
+A DN typically looks like `CN=MyUser,CN=Users,DC=mydomain,DC=com`
+although the exact sequence of tokens depends on the LDAP server
+configuration.
+
+It can be any valid DN as long as LDAP allows this user to query data.
+
+This configuration is also useful when you are connecting to Active
+Directory from a Unix machine, as AD doesn't allow anonymous bind by
+default. But if you can't figure this out, you can also change AD
+setting to allow anonymous bind. 
+
+### Disable LDAP Email resolver
+
+Controls whether LDAP will be used to try and resolve the email
+addresses of users.
+
+### Enable cache
+
+Some LDAP servers may be slow, or rate limit client requests.
+
+In such cases enabling caching may improve performance of Jenkins with
+the risk of delayed propagation of user changes from LDAP and increased
+memory usage on the master.
+
+Note: The default configuration is to leave the cache turned off.
+
+### Environment Properties
+
+As of 1.7 of the LDAP plugin, you can now specify additional Environment
+Properties to provide the backing Java LDAP client API. See [Oracle's
+documentation](http://docs.oracle.com/javase/7/docs/technotes/guides/jndi/jndi-ldap.html){.external-link} for
```

Comment:
Use Java 8 reference documentation.  Java 7 no longer supported with Jenkins
```suggestion
documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/jndi/jndi-ldap.html){.external-link} for
```

Reply:
Should I also remove the `{external.link}`

Reply:
yes